### PR TITLE
Publish documentation on npmjs and massa server

### PR DESCRIPTION
**Pull request by Elli610**

### CODE REVIEW

**Review by BatiGencho — APPROVED** [review]

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Publish doc
+
+on:
+  push:
+    branches: 
+      - master
```
[the remark this answers is not in the document]
Comment (gregLibert) [reply .github/workflows/npm-publish.yml:6]: Doc should only be publish on new tag. have a look at massa-as-sdk to fix this

File: .github/workflows/npm-publish.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Publish doc
```
Comment (gregLibert) [inline .github/workflows/npm-publish.yml:1]: It not only about doc, but also NPM

File: .github/workflows/npm-publish.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.npm_token }}
+
+  deploy-typedoc:
+      runs-on: ubuntu-latest
+      steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
Comment (gregLibert) [inline .github/workflows/npm-publish.yml:32]: should use LTS (18) version

File: README.md [hunk 4]
```diff
@@ -177,5 +177,4 @@ await  myProvider.importAccount(myAccount.address());
 ## Contributing
 We welcome contributions from the community!
 
-If you would like to contribute to Massa-as-sdk, please read the  [CONTRIBUTING file](https://github.com/massalabs/wallet-provider/blob/main/CONTRIBUTING.md).
-  
+If you would like to contribute to Massa-as-sdk, please read the  [CONTRIBUTING file](https://github.com/massalabs/wallet-provider/blob/main/CONTRIBUTING.md).
```
[the remark this answers is not in the document]
Comment (gregLibert) [reply README.md:180]: But this is not linked to this issue. next time create another PR for this

File: src/account/AccountSign.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ export interface IAccountSignRequest {
 }
 
 /**
- * This interface represents the response object containing the signed message data, which is sent by the content script 
+ * This interface represents the response object containing the signed message data, which is sent by the content script
```
Comment (gregLibert) [inline src/account/AccountSign.ts:10]: Those are linter fix. they should be in another PR. Plus we should have a CI checking all that. Can you create an issue for the CI to check the linter ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: .github/workflows/npm-publish.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Elli610) [reply .github/workflows/npm-publish.yml:6]: Done

File: src/account/AccountSign.ts (hunk 5 above)
Comment (Elli610) [reply src/account/AccountSign.ts]: done

File: .github/workflows/npm-publish.yml (hunk 3 above)
Comment (Elli610) [reply .github/workflows/npm-publish.yml:32]: done

File: .github/workflows/npm-publish.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  publish-npm:
+    name: publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
```
[the remark this answers is not in the document]
Comment (Elli610) [reply .github/workflows/npm-publish.yml:19]: done

**Review by gregLibert — APPROVED** [review]

**Pull request merged**